webapi/src/main/scala/org/knora/webapi/store/triplestore/upgrade/plugins/UpgradePluginPR2079.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright © 2021 - 2022 Swiss National Data and Service Center for the Humanities and/or DaSCH Service Platform contributors.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.knora.webapi.store.triplestore.upgrade.plugins
+
+import com.typesafe.scalalogging.Logger
+import org.knora.webapi.feature.FeatureFactoryConfig
+import org.knora.webapi.messages.OntologyConstants
+import org.knora.webapi.messages.util.rdf._
+import org.knora.webapi.store.triplestore.upgrade.UpgradePlugin
+
+/**
+ * Transforms a repository for Knora PR 2079.
+ * Adds missing datatype ^^<http://www.w3.org/2001/XMLSchema#anyURI> and/or value to valueHasUri
+ */
+class UpgradePluginPR2079(featureFactoryConfig: FeatureFactoryConfig, log: Logger) extends UpgradePlugin {
+  private val nodeFactory: RdfNodeFactory = RdfFeatureFactory.getRdfNodeFactory(featureFactoryConfig)
+
+  override def transform(model: RdfModel): Unit = {
+    val statementsToRemove: collection.mutable.Set[Statement] = collection.mutable.Set.empty
+    val statementsToAdd: collection.mutable.Set[Statement]    = collection.mutable.Set.empty
+    val newObjectValue: String => DatatypeLiteral = (value: String) =>
+      nodeFactory.makeDatatypeLiteral(value, OntologyConstants.Xsd.Uri)
+
+    for (statement: Statement <- model) {
```

Comment:
this solution with a for-loop and two mutable collections seems very non-functional 😉

Reply:
Feel free to refactor it mate 💪